extensions/oidc/runtime/src/main/java/io/quarkus/oidc/runtime/DefaultTenantConfigResolver.java

```diff
@@ -320,6 +306,50 @@ public JavaScriptRequestChecker getJavaScriptRequestChecker() {
         return javaScriptRequestChecker.isResolvable() ? javaScriptRequestChecker.get() : null;
     }
 
+    private static TenantResolver[] prepareStaticTenantResolvers(TenantConfigBean tenantConfigBean, String rootPath,
+            Instance<TenantResolver> tenantResolverInstance, boolean resolveTenantsWithIssuer,
+            TenantResolver defaultStaticTenantResolver) {
+        List<TenantResolver> staticTenantResolvers = new ArrayList<>();
+        // STATIC TENANT RESOLVERS BY PRIORITY:
+        // 0. annotation based resolver
+
+        // 1. custom tenant resolver
+        if (tenantResolverInstance.isResolvable()) {
+            if (tenantResolverInstance.isAmbiguous()) {
+                throw new IllegalStateException("Multiple " + TenantResolver.class + " beans registered");
+            }
+            staticTenantResolvers.add(tenantResolverInstance.get());
+        }
+
+        // 2. path-matching tenant resolver
+        var pathMatchingTenantResolver = PathMatchingTenantResolver.of(tenantConfigBean.getStaticTenantsConfig(), rootPath,
+                tenantConfigBean.getDefaultTenant());
+        if (pathMatchingTenantResolver != null) {
+            staticTenantResolvers.add(pathMatchingTenantResolver);
+        }
+
+        // 3. default static tenant resolver
+        if (!tenantConfigBean.getStaticTenantsConfig().isEmpty()) {
+            staticTenantResolvers.add(defaultStaticTenantResolver);
+        }
+
+        // 4. session-based tenant resolver
```

Comment:
This property check is like a fallback, wrapping it in TenantResolver is making it harder to notice  (and it can also be set by the state cookie). I'd not mind running this check after all others (may ne ), incl the issuer one is run - realistically, the issuer based one won't be in the array for web-apps.

I see right now it is checked twice on `main`:
```
private String resolveStaticTenantId(RoutingContext context) {
        String tenantId = context.get(OidcUtils.TENANT_ID_ATTRIBUTE);
        if (isTenantSetByAnnotation(context, tenantId)) {
            return tenantId;
        }

        // all the other checks

        // finally
        if (tenantId == null) {
            tenantId = context.get(OidcUtils.TENANT_ID_ATTRIBUTE);
        }
        return tenantId;
    }
```

so may be we can do
```
private String resolveStaticTenantId(RoutingContext context) {
        String resolvedTenantId = context.get(OidcUtils.TENANT_ID_ATTRIBUTE);
        if (isTenantSetByAnnotation(context, resolvedTenantId)) {
            return resolvedTenantId;
        }

        // all the other checks which will return if tenantId is not null

        // finally
        return resolvedTenantId;
    }
```

this will make it easier to add `resolvedTenantId` in between where necessary as opposed to fixing it in the array.

hope it can work for you

Reply:
I don't exactly feel it's harder to mention, but I think suggestion is perfectly fine. I'll return it. Thanks

Reply:
What calms my heart is removing it turned out to avoid one null check and the code looks nice. Done.